src/MusicPlaylistMVP.java

```diff
@@ -0,0 +1,155 @@
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Represents a song with a name and an artist.
+ */
+class Song {
```

Comment:
I am not sure Java lets you put two classes in the same file. Maybe in newer versions this is possible, but Java historically only allows one class per file (otherwise, they need to be nested).